### DISCUSSION TITLE
custom/local @ folders, “hide” list

### DISCUSSION THREAD

**Discussion by the-steech** [ppooll-dev/ppooll#67] [Ideas]
ppooll_contributions folder has got me thinking about adding a setting/button for additional act folders.  this might help separate acts into "families" within ppooll_contributions, such as Max for live, efx, and ondo ports.  (Maybe even saving these patches with names such as "m4l.romb", "ondo.djrone", etc)

Additionally, a "hide" list could be helpful for just adding names of acts that you don't want to see in the main act-menu

**Comment by the-steech** [comment]
Also a "legacy" folder would be nice :)

**Comment by klausfilip** [comment]
yes this has to come sooner or later. somthing like in act_overview, but another category of tags ?

**Comment by chausch** [comment]
maybe subfolders in acts would do?

**Reply by klausfilip** [reply]
hmm, how to define these subfolders ?
another category in our act_overview would do better.
and is also more flexible..
anyway, i should have put the json like this::
{
	"acts" : 	{
		"2Dsliders" : 		{
			"description" : "control parameters by dragging points",
			"authors" : 1,
			"tags" : 4
		}
,
		"INmulti" : 		{
			"description" : "(multiple) audio input",
			"authors" : 1,
			"tags" : [ 2, 13, 15 ]
		}
,
		"LFFO" : 		{
			"description" : "dynamic ring modulator",
			"authors" : 4,
			"tags" : [ 9, 15 ]
		}
,

**Reply by chausch** [reply]
+1 to the more "verbose" json structure!

**Comment by chausch** [comment]
regarding ondo-ports: i don't think we should categorize acts by author or algorithm-creator.
rather i'd suggest using a classification based on area of application (as in spat.***).
straight-up ports with no additional code contribution could theoretically go in a ports category though. but i think acts like that could maybe just go into contributions branch instead.

**Comment by klausfilip** [comment]
what about a second act.json file that adds user-costomizable tags ?
like favorites, hide, etc
or add it to "preferences" and make a json out of this as well

**Reply by chausch** [reply]
if the users could manage to fill these customizations from the ppooll ui this would be an awesome solution! 
(but if it would require tinkering with a text editor it would probably too complicated for most users.)

**Reply by chausch** [reply]
maybe an option in the act menu to add user tags (per act)?

so open an act, select the menu, select tag edit, choose "my super synth selection" and the act gets displayed in a corresponding section in acts menu.

like this?

**Reply by klausfilip** [reply]
just pushed my act_overview update. cx you may test the new edit function for "tags"

**Reply by chausch** [reply]
that's really cool! however, maybe this feature should only be available for the extra customizable list you proposed above for sorting acts? 🤔 maybe this could be saved with the ppooll_preferences file!

**Comment by the-steech** [comment]
regarding custom (user) @ folders, i think i am going to try to abstract the act_menu "append" actions to support a list of directories:  

DEFAULTS (will always we be searched):
/Max 8/Packages/ppooll/patches/ppooll.acts 
/Max 8/Packages/ppooll_contributions/patches/ppooll.acts

and then maybe something like this:
/Max 8/Library/ppooll_user/patches/ppooll.acts

**Comment by klausfilip** [comment]
hmm, if the folder has a unique name, the user can put it anywhere. with your suggestion, the user has to use the correct foldername and the correct path.

**Comment by the-steech** [comment]
it could be "hard-coded" to the path i described, but ideally it would work with an "opendialog" object so the user can open a folder thru Mac Finder or Win Explorer.  if that's what you're saying @klausfilip, i totally agree.

**Comment by klausfilip** [comment]
ok, thats a way: again at the very end of the act-menu i will add sth like "own act folder" which opens a file-dialog.
the path gets stored then in ppooll-preferences.

**Reply by the-steech** [reply]
amazing !!!

**Comment by the-steech** [comment]
created a branch called [multiple-act-paths](https://github.com/ppooll-dev/ppooll/tree/multiple-act-paths) which abstracts some of the logic for parsing filepaths and labels.  that way a user can add as many directories as they want.

(the reason i created a separate branch here is cause i figure you can review and lmk if i'm headed in the right direction :) )

**Comment by klausfilip** [comment]
hi joe, @the-steech 
i only realize now, that you did start exactly the work, that i was doing in moving the preferences to json and the act-menu...
stupid me, sorry or that !!!

**Reply by the-steech** [reply]
no worries, you're implementation is great!  i was trying to abstract it so you could have a whole list of folder, but i actually think limiting to one "custom/unshared" folder is much simpler :)  but the subpatch I created to streamline is here in case we ever need to go back

**Comment by Pogslammer** [comment]
Hey there!

Is there any place I can download the default ppooll-preferences.json file? I was trying to edit it to put my favorite list in alphabetical order and I seem to have messed it up (I don't know what I'm doing). It's just blank now and the add/delete favorite functions don't work anymore. I tried reinstalling the latest code, but that's not doing it. Sorry about that, any guidance appreciated!

**Comment by klausfilip** [comment]
hi @Pogslammer 
well, if you remove your current ppooll-preferences.json from the ppooll_presets folder, it will be created as default there again.
(hopefully)
after that try to edit your favorites with _act_overview, this will keep the alphabetical order

**Comment by klausfilip** [comment]
also noted: adding favorites from its menu should keep alphabetical order ;)

**Reply by Pogslammer** [reply]
Yes! That did it, and all that is working beautifully, thanks so much for your help. This is great.
